modules/tiles/src/tileset/tile-3d.ts

```diff
@@ -327,14 +327,19 @@ export default class TileHeader {
       // The content can be a binary tile ot a JSON tileset
       const loader = this.tileset.loader;
       const options = {
-        fetch: this.tileset.fetchOptions,
         [loader.id]: {
           isTileset: this.type === 'json',
           ...this._getLoaderSpecificOptions(loader.id)
-        }
+        },
+        ...this.tileset.loadOptions
       };
 
       this.content = await load(contentUrl, loader, options);
+
+      if (this.tileset.options.contentLoader) {
```

Comment:
We already have an `onTileLoad` callback. Do we need to add another callback? If we do, can we follow the convention and call it `onTileContentLoad` or similar?

Reply:
This is not the same as the `onTile`* callbacks which are essentially events.
The content loader promise must resolve before a tile can set 
```
      this.contentState = TILE_CONTENT_STATE.READY;
      this._onContentLoaded();
```
And only then those callbacks may be fired.
Without this solution there is no way to use `async` loaders such as the three.js `GLTFLoader` without experiencing a race condition in which a tile is selected before it is actually on the render graph.

Reply:
Understood, thanks for clearing up the use case. 

Would it make sense if we could declare it as `onTileContentLoad(tile: Tile3D): Promise<void>` and we wait for it?

Reply:
I am a bit hesitant because usually `on*` callbacks are not something you would expect would block execution. But I defer to you on this one.

Reply:
Yes I can see your point. Thoughts:
- `contentLoader` risk being confusing in a framework that is specifically built around a concept of "loaders". I would have expected an option with that name to accept a loaders.gl loader.
- Perhaps another name, ideally indicating some new naming convention?
- We could also allow all the `on...` functions to return promises though this might lead to unintended synchronization for current users, if they are using async functions as callbacks.

When working with a big documented API the API audit tends to be a big part of the effort. This is why it is often best to make one PR for every addition, instead of one PR with 4 new props :)

Reply:
Perhaps `tileProcessingHook` or something of sort?
Understood about the heavy PRs. Will work to make those leaner in the future.